### PR TITLE
Updated README: Release Notes and Citations.

### PR DESCRIPTION
**Pull request by dwelsch-esi**
- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

### CODE REVIEW

**Comment by dwelsch-esi** [comment]
@santoshkumarradha Changes from your last comments checked in. Please review.

File: README.md [hunk 1]
```diff
@@ -13,226 +13,100 @@
 
 </div>
 
-## 🤔 What is Covalent?
+<img src="https://raw.githubusercontent.com/AgnostiqHQ/covalent/master/doc/source/_static/executor_gif_full_list_tall.gif" align="right" width="40%" alt="Covalent Executors"></img>
```
[the remark this answers is not in the document]
Comment (santoshkumarradha) [reply README.md:16]: hey @dwelsch-esi I would add make it this one.

```suggestion
<p align="center">
<img src="./doc/source/_static/executor_gif_full_list_tall.gif" width="70%" alt="Covalent Executors" style="border-radius:10%"></img> 
</p>
```

File: README.md [hunk 2]
```diff
@@ -13,226 +13,100 @@
...
 
-Covalent is a Pythonic workflow tool used to execute HPC and quantum tasks in heterogenous environments. Computational scientists and engineers use Covalent to...
+## What is Covalent?
```
Comment (santoshkumarradha) [inline README.md:18]:
```suggestion
## 🤔  What is Covalent?
```

File: README.md [hunk 3]
```diff
@@ -13,226 +13,100 @@
...
-- visualize data and task dependencies in an interactive user interface
-- run code in heterogenous compute environments, including in hybrid-cloud and hybrid-quantum configurations
-- understand where time and money is spent across a project
+Covalent is a Pythonic workflow tool for computational scientists, AI/ML software engineers, and anyone who needs to run experiments on limited or expensive computing resources including quantum computers, HPC clusters, GPU arrays, and cloud services.
 
-Covalent may be deployed locally or on a remote server. Covalent is rapidly expanding to include support for a variety of cloud interfaces, including HPC infrastructure tools developed by major cloud providers and emerging quantum APIs. It has never been easier to deploy your code on the world's most advanced computing hardware with Covalent.
+Covalent enables a researcher to run computation tasks on an advanced hardware platform – such as a quantum computer or serverless HPC cluster – using a single line of code.
 
-Read more in the official [documentation](https://covalent.readthedocs.io/en/latest/).
+## Why Covalent?
```
Comment (santoshkumarradha) [inline README.md:24]:
```suggestion
## 💭 Why Covalent?
```

**Review by santoshkumarradha — APPROVED** [review]

**Comment by santoshkumarradha** [comment]
@AlejandroEsquivel can you help in formatting the gif and merging this ? TIA ! 🤝

**Comment by dwelsch-esi** [comment]
@AlejandroEsquivel Please ping me on Slack - @Dave Welsch - if you have any questions about the PR. In fact, ping me anyway so I'll have your Slack info. 😄

**Review by santoshkumarradha — DISMISSED** [review]

**Pull request merged**